roles/mongodb/tasks/main.yml

```diff
@@ -1,25 +1,19 @@
 ---
-- name: Add mongodb key
-  become: yes
-  apt_key:
-    keyserver: keyserver.ubuntu.com
-    id: 42F3E95A2C4F08279C4960ADD68FA50FEA312927
-    state: present
+- name: Install mongodb on {{ ansible_distribution }}
+  include: mongodb_{{ ansible_pkg_mgr }}.yaml
   tags: [databases, mongodb]
 
-- name: Add mongodb repository
+- name: Start & Enable mongodb
   become: yes
-  apt_repository:
-    repo: 'deb http://repo.mongodb.org/apt/{{ ansible_distribution|lower }} {{ ansible_distribution_release|lower }}/mongodb-org/3.2 multiverse'
-    state: present
+  service:
+    name: mongod
+    state: started
+    enabled: yes
   tags: [databases, mongodb]
 
-- name: Install mongodb
+- name: chkconfig mongod on | el 6 only
   become: yes
-  apt:
-    name: mongodb-org
-    update_cache: yes
-    state: present
-  notify:
-    - restart mongodb
+  command: chkconfig mongod on
```

Comment:
> From discussion: https://stackstorm.slack.com/archives/ops/p1483720827000240

After taking a look, seems lke Ansible `service` module handles everything good enough even for EL6 (`3` and `5` runlevels). 
Just double check if `mongod` really worked on startup for you without explicitly calling `chkconfig` and if so this block could be safely removed.